pkg/api/devops/v1alpha2/types.go

```diff
@@ -0,0 +1,20 @@
+package v1alpha2
+
+import "time"
+
+type PageableDevOpsProject struct {
```

Comment:
What is the purpose of this change?

Reply:
currently, we create a client instance through client factory,  so import path will become
```
kubesphere.io/kubesphere/pkg/simple/client/devops -> kubesphere.io/kubesphere/pkg/simple/client
kubesphere.io/kubesphere/pkg/simple/client/kubesphere -> kubesphere.io/kubesphere/pkg/simple/client
```
In kubesphere client, https://github.com/kubesphere/kubesphere/blob/master/pkg/simple/client/kubesphere/kubesphereclient.go#L48 , this will cause cyclic import. 

Plus, it's better organized if we place types definition in a independent package, and we can avoid potential cyclic import.